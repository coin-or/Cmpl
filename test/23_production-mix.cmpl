parameters:
	price[] := (500, 600, 450, );
	costs[] := (425, 520, 400,);

	#machine hours required per unit
	a[,] := ((8, 15, 12), (15, 10, 8));

	#upper bounds of the machines
	b[] := (1000, 1000);

	#profit contribution per unit
	{j in defset(price):  c[j] := price[j]-costs[j]; }

	#upper bound of the products
	xMax[] := (250, 240, 250 );

variables:
	x[ defset(price)]: integer; 

objectives:
	profit: c[]T  * x[] ->max;

constraints:
	res: a[,] * x[] <= b[];
	0<=x[]<=xMax[];
 
