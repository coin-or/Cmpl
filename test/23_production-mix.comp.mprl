<?xml version="1.0" encoding="UTF-8"?>
<mprl xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://www.coliop.org/schemas/Mprl.xsd">
	<general>
		<generalStatus>normal</generalStatus>
		<instanceName>23_production-mix.cmpl</instanceName>
		<mplName>CMPL</mplName>
		<message>cmpl finished normal</message>
	</general>
</mprl>