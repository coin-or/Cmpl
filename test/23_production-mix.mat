Variable name                      x[1]           x[2]           x[3]
Variable type                         I              I              I

profit              max              75             80             50
   Subject to                                                                    RHS
res_1               L                 8             15             12           1000
res_2               L                15             10              8           1000

Lower Bound                           0              0              0
Upper Bound                         250            240            250
