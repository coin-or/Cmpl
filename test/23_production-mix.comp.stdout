---------------------------------------------------------------------------------------------------------
General status:     normal
Problem :           23_production-mix.cmpl
SolverName          COIN-OR cbc
Nr. of Solutions:   1
---------------------------------------------------------------------------------------------------------
Solution nr.:       1
Objective name :    profit
Objective value :   6395 (max!)

Variables           
Nr. of variables:   3
Name                 Type            Activity         Lower Bound         Upper Bound            Marginal
---------------------------------------------------------------------------------------------------------
x[1]                    I                  33                   0                 250                   -
x[2]                    I                  49                   0                 240                   -
x[3]                    I                   0                   0                 250                   -
---------------------------------------------------------------------------------------------------------

Constraints         
Nr. of constraints: 2
Name                 Type            Activity         Lower Bound         Upper Bound            Marginal
---------------------------------------------------------------------------------------------------------
res_1                   L                 999           -Infinity                1000                   -
res_2                   L                 985           -Infinity                1000                   -
---------------------------------------------------------------------------------------------------------
